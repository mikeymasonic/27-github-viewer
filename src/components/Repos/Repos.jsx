import React from 'react';
import PropTypes from 'prop-types';
import Repo from '../Repo/Repo.jsx';

const Repos = ({ repos }) => {
  const repoMap = repos.map(repo => <Repo key={repo.id} {...repo} />);
  return (
    <ul>
      {repoMap}
    </ul>
  );
};

Repos.propTypes = {
  repos: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    html_url: PropTypes.string.isRequired
  })).isRequired
};

export default Repos;
