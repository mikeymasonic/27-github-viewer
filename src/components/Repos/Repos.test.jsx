import React from 'react';
import { shallow } from 'enzyme';
import Repos from './Repos';

describe('Repos component', () => {
  it('renders Repos', () => {
    const repos = [
      {
        id: 235044203,
        name: 'capTouch-Fader-Joystick-Midi-Device',
        html_url: 'https://github.com/mikeymasonic/capTouch-Fader-Joystick-Midi-Device'
      },
      {
        id: 247767043,
        name: 'arcade-button-midi-controller',
        html_url: 'https://github.com/mikeymasonic/arcade-button-midi-controller'
      }
    ];
    const wrapper = shallow(<Repos repos={repos}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
