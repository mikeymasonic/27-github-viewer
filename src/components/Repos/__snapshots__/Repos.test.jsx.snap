// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repos component renders Repos 1`] = `
<ul>
  <Repo
    html_url="https://github.com/mikeymasonic/capTouch-Fader-Joystick-Midi-Device"
    id={235044203}
    key="235044203"
    name="capTouch-Fader-Joystick-Midi-Device"
  />
  <Repo
    html_url="https://github.com/mikeymasonic/arcade-button-midi-controller"
    id={247767043}
    key="247767043"
    name="arcade-button-midi-controller"
  />
</ul>
`;
